ehrapy/plot/_sankey.py

```diff
@@ -0,0 +1,180 @@
+    labels = list(dict.fromkeys(labels))  # keep order & unique
+
+    # Build links between consecutive columns
+    sources, targets, values = [], [], []
+    source_levels, target_levels = [], []
+    for i in range(len(columns) - 1):
+        col_from, col_to = columns[i], columns[i + 1]
+        flows = df.groupby([col_from, col_to]).size().reset_index(name="count")
+        for _, row in flows.iterrows():
+            source = f"{col_from}: {row[col_from]}"
+            target = f"{col_to}: {row[col_to]}"
+            sources.append(source)
+            targets.append(target)
+            values.append(row["count"])
+            source_levels.append(col_from)
+            target_levels.append(col_to)
+
+    sankey_df = pd.DataFrame(
+        {
+            "source": sources,
+            "target": targets,
+            "value": values,
+            "source_level": source_levels,
+            "target_level": target_levels,
+        }
+    )
+
+    sankey = hv.Sankey(sankey_df, kdims=["source", "target"], vdims=["value"])
+    default_opts = {"label_position": "right", "show_values": True, "title": f"Flow across: {columns}"}
+
+    default_opts.update(kwargs)
+
+    sankey = sankey.opts(opts.Sankey(**default_opts))
+
+    if show:
+        from IPython.display import display
+
+        display(sankey)
+
+    return sankey
+
+
+def plot_sankey_time(
+    edata: EHRData,
+    *,
+    columns: list[str],
+    layer: str,
+    state_labels: dict[int, str] | None = None,
+    show: bool = False,
+    **kwargs,
+) -> hv.Sankey:
+    """Create a Sankey diagram showing patient state transitions over time.
+
+    This function visualizes how patients transition between different states
+    (e.g., disease severity, treatment status) across consecutive time points.
+    Each node represents a state at a specific time point, and flows show the
+    number of patients transitioning between states.
+
+    Please call :func:`holoviews.extension` with ``"matplotlib"`` or ``"bokeh"`` before using this function to select the backend.
+
+    Args:
+        edata: Central data object containing observation data
+        columns: Column names from edata.var_names to visualize
+        layer: Name of the layer in `edata.layers` containing the feature data to visualize.
+        state_labels: Mapping from numeric state values to readable labels. If None, state values
+                    will be displayed as strings of their numeric codes (e.g., "0", "1", "2"). Default: None
+        show: If True, display the plot immediately. If False, only return the plot object without displaying.
+        **kwargs: Additional styling options passed to `holoviews.opts.Sankey`. See HoloViews Sankey documentation for full list of options.
+
+    Returns:
+        holoviews.Sankey
+
+    Examples:
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> import ehrdata as ed
+    >>>
+    >>> layer = np.array(
```

Comment:
This example is too complex. Can we make this work with blobs? If not, can we adapt the blobs function so that you can use it here, please?

Reply:
Yeah let's maybe add an option then